src/minimap.cpp

```diff
@@ -0,0 +1,476 @@
+
+	porting::setThreadName("MinimapUpdateThread");
+
+	while(!StopRequested()) {
+
+		while(m_queue.size()) {
+			QueuedMinimapUpdate *q = m_queue.pop();
+			std::map<v3s16, MinimapMapblock *>::iterator it;
+			it = m_blocks_cache.find(q->p);
+			if (q->data)
+				m_blocks_cache[q->p] = q->data;
+			else if (it != m_blocks_cache.end())
+				m_blocks_cache.erase(it);
+		}	
+
+		if (data->map_invalidated) {
+			if (data->mode != 0) {
+				getMap(data->pos, data->map_size, data->scan_height, data->radar);
+				data->map_invalidated = false;
+			}
+		}
+		sleep_ms(10);
+	}
+	END_DEBUG_EXCEPTION_HANDLER(errorstream)
+
+	return NULL;
+}
+
+MinimapPixel* MinimapUpdateThread::getMinimapPixel (v3s16 pos, s16 height, s16 &pixel_height)
+{
+	pixel_height = height - MAP_BLOCKSIZE;
+	v3s16 blockpos_max, blockpos_min, relpos;
+	getNodeBlockPosWithOffset(v3s16(pos.X, pos.Y - height / 2, pos.Z), blockpos_min, relpos);
+	getNodeBlockPosWithOffset(v3s16(pos.X, pos.Y + height / 2, pos.Z), blockpos_max, relpos);
+	std::map<v3s16, MinimapMapblock *>::iterator it;
+	for (s16 i = blockpos_max.Y; i > blockpos_min.Y - 1; i--) { 
+		it = m_blocks_cache.find(v3s16(blockpos_max.X, i, blockpos_max.Z));
+		if (it != m_blocks_cache.end()) {
+			MinimapPixel *pixel = &it->second->data[relpos.X + relpos.Z * MAP_BLOCKSIZE];
+			if (pixel->id != CONTENT_AIR) {
+				pixel_height += pixel->height;
+				return pixel;
+			}
+		}
+		pixel_height -= MAP_BLOCKSIZE;
+	}
+	return NULL;
+}
+
+s16 MinimapUpdateThread::getAirCount (v3s16 pos, s16 height)
+{
+	s16 air_count = 0;
+	v3s16 blockpos_max, blockpos_min, relpos;
+	getNodeBlockPosWithOffset(v3s16(pos.X, pos.Y - height / 2, pos.Z), blockpos_min, relpos);
+	getNodeBlockPosWithOffset(v3s16(pos.X, pos.Y + height / 2, pos.Z), blockpos_max, relpos);
+	std::map<v3s16, MinimapMapblock *>::iterator it;
+	for (s16 i = blockpos_max.Y; i > blockpos_min.Y - 1; i--) {
+		it = m_blocks_cache.find(v3s16(blockpos_max.X, i, blockpos_max.Z));
+		if (it != m_blocks_cache.end()) {
+			MinimapPixel *pixel = &it->second->data[relpos.X + relpos.Z * MAP_BLOCKSIZE];
+			air_count += pixel->air_count;
+		}
+	}
+	return air_count;
+}
+
+void MinimapUpdateThread::getMap (v3s16 pos, s16 size, s16 height, bool radar)
+{
+	v3s16 p = v3s16 (pos.X - size / 2, pos.Y, pos.Z - size / 2);
+
+	for(s16 x = 0; x < size; x++) {
+		for(s16 z = 0; z < size; z++){
+			u16 id = CONTENT_AIR;
+			MinimapPixel* minimap_pixel = &data->minimap_scan[x + z * size];
+			if (!radar) {
+				s16 pixel_height = 0;
+				MinimapPixel* cached_pixel = 
+					getMinimapPixel(v3s16(p.X + x, p.Y, p.Z + z), height, pixel_height);
+				if (cached_pixel) {
+					id = cached_pixel->id;
+					minimap_pixel->height = pixel_height;
+				}
+			} else {
+				minimap_pixel->air_count = getAirCount (v3s16(p.X + x, p.Y, p.Z + z), height);
+			}
+			minimap_pixel->id = id;
+		}
+	}
+}
+
+Mapper::Mapper(IrrlichtDevice *device, Client *client):
+	m_minimap_update_thread(MinimapUpdateThread(device, client))
```

Comment:
/home/nerzhul/Devel/minetest/src/minimap.cpp:211:2: error: call to implicitly-deleted copy constructor of 'MinimapUpdateThread'
        m_minimap_update_thread(MinimapUpdateThread(device, client))

Clang doesn't like it